docs/exit_validation.md

```diff
@@ -87,3 +105,88 @@ All the general rules will apply in the MoreVP world.
 Invalid attempts to do an exiting action using MoreVP must excite challenges.
 Absence of challenges within some period (like `sla_margin`) must result in client prompting to exit.
 `State` will be modified on finalization, and if the finalization is invalid should `:invalid_finalization` and prompt exit.
+
+## In-flight exits
+
+With MoreVP, we need to handle another type of exit game, which is the in-flight exit game, as specced out [here](docs/morevp.md).
+
+In terms of handling within the Watcher, similar principles will apply:
+  - we gather and keep in `OMG.Watcher.ExitProcessor`'s persistent state the current state of in-flight txs and exits
+  - we periodically check the validity of this state, emitting events and allowing for actions as necessary
+  - we touch `OMG.API.State` only when the in-flight exit finalizes
+  - if some invalid IFE or piggyback runs unchallenged for too long, it should be a prompt to exit (ala `:unchallenged_exit` above)
+
+**[Diagram](https://docs.google.com/drawings/d/1UaAMZTJBbikTM0eFSbNM7ZVgi1HZrLTfwYCUvHGUaW0/edit?usp=sharing)** illustrates the flows described below (with the addition of handling on the Child Chain Server side for comparison).
```

Comment:
Minor problem with "spend all inputs" block on ch-ch side.
tx1 = [[A, B],[C]], honest spend by A, not included somehow. IFE by B.
tx2 = [[B],[D]], double spend by B. IFE by B.
Now A can be only piggybacked, it can't be spend on plasma chain anymore because of "spend all inputs".

Reply:
There is an attack vector which opens with "outputs mine" block in the watcher side of the diagram. Watcher also needs to check if inputs are in correct block which it has access to. Only in such case outputs can be exited at all.

If watcher does not perform such check, operator can use it as a part of his own DoS of Ethereum. By creating IFE [[input_from_withhold_block],[[A1],[A2],[A3],[A4]]] operator can provoke four Alices to do a piggyback, amplifying it's own gas investment, creating a DDoS out of his own DoS.

Reply:
There is an attack vector which opens with "outputs mine" block in the watcher side of the diagram. Watcher also needs to check if inputs are in correct block which it has access to. Only in such case outputs can be exited at all.

If watcher does not perform such check, operator can use it as a part of his own DoS of Ethereum. By creating IFE [[input_from_withhold_block],[[A1],[A2],[A3],[A4]]] operator can provoke four Alices to do a piggyback, amplifying it's own gas investment, creating a DDoS out of his own DoS.

Reply:
but A has had the intention to spend, so ch-ch is (in a way) justified to drop the `A` from spendable. `B` is attacking `A` in this situation, and will get punished with at least one `exit_bond`.

not sure how sever it is. An alternative which would potentially heal this is when `spend all inputs` happens only after the Phase 1 passes, and the tx is deemed canonical. But that might open another can of worms.

Reply:
pasting a richer description by @paulperegud: (let's continue here)

> 
> There is an attack vector which opens with "outputs mine" block in the watcher side of the diagram. Watcher also needs to check if inputs are in correct block which it has access to. Only in such case outputs can be exited at all.
> 
> If watcher does not perform such check, operator can use it as a part of his own DoS of Ethereum. By creating IFE [[input_from_withhold_block],[[A1],[A2],[A3],[A4]]] operator can provoke four Alices to do a piggyback, amplifying it's own gas investment, creating a DDoS out of his own DoS.